### ISSUE TITLE
[miral-shell] Tiling sometimes fails to resize window when becoming focussed

### ISSUE THREAD

**Issue by RAOF** [MirServer/mir#2325]
With a mix of xterms, qterminal, and kate windows open, *sometimes* the focused window will fail to resize, and there will be a little wedge window (same size as the other inactive windows on the right) in the focused, left position.

**Comment by AlanGriffiths** [comment]
I don't think this is a blocker for the release, as it is likely the example code that's a problem. (Or, more significantly, the miral API needs a significant rework to better synchronize window management and communication with the client.)

I think I've seen this occasionally in the past. Never had a reliable reproducer to help track it down. (The WAYLAND_DEBUG and --widow-management-trace logs would be helpful, but not captured this problem on them yet.)
